src/core/execute.c

```diff
@@ -4014,6 +4014,7 @@ static int exec_parameters_get_cgroup_path(const ExecParameters *params, char **
          * flag, which is only passed for the former statements, not for the latter. */
 
         using_subcgroup = FLAGS_SET(params->flags, EXEC_CONTROL_CGROUP|EXEC_CGROUP_DELEGATE|EXEC_IS_CONTROL);
+        using_subcgroup &= cg_all_unified() > 0;
```

Comment:
nah, please don't. we don#t want to deviate needlessy for cgroupsv1 on this. cgroupsv1 is in maintainance mode anyway, and optimizing it is not in focus for us. Let's arrange things everywhere the same.

after all regardless if you are in cgroupsv1 or cgroupsv2 mode, it's not conceptually clear what it means to schedule a process against a cgroup, hence the "no-processes-in-inner-cgroups" rule makes sense either way.

Reply:
I changed my mind and it's perhaps simpler to not special-case v1 instead of explicitly disallowing it for the subcgroup.
(Although the scheduling argument is weakened by touching only controller-less hierarchies on v1.)